oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2TokenIntrospectionResponse.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core.endpoint;
+
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.toMap;
+import static org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames.CLIENT_ID;
+import static org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames.SCOPE;
+import static org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames.TOKEN_TYPE;
+import static org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames.USERNAME;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.AUD;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.EXP;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.IAT;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.ISS;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.JTI;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.NBF;
+import static org.springframework.security.oauth2.jwt.JwtClaimNames.SUB;
+
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken.TokenType;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A representation of an OAuth 2.0 Introspection Token Response.
+ *
+ * @author Gerardo Roza
+ * @since 0.1.1
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7662#section-2.2">Section 2.2 - Introspection Response</a>
+ */
+public class OAuth2TokenIntrospectionResponse {
```

Comment:
I don't think we need this class at this point. Please try to remove it.

Reply:
Actually, on second thought, we do need this. However, please rename to `OAuth2TokenIntrospectionClaims` and move to package `org.springframework.security.oauth2.core` - this will be used on client side as well.

Also, please follow the design of `OidcProviderConfiguration` so you should also add `OAuth2TokenIntrospectionClaimAccessor`.

Reply:
Also please move `OAuth2TokenIntrospectionResponseHttpMessageConverter` to package `org.springframework.security.oauth2.core.http.converter`

Reply:
Cool, I now did all the changes @jgrandja . Just a heads up regarding the `OAuth2TokenIntrospectionClaimAccessor`, instead of defining and using a `OAuth2IntrospectionClaimNames` class as in the Resource Server version (`OAuth2IntrospectionClaimNames.SCOPE`, `OAuth2IntrospectionClaimNames.TOKEN_TYPE`, `OAuth2IntrospectionClaimNames.EXPIRES_AT`,...), I decided to use the definitions we already have (`OAuth2ParameterNames`, `JwtClaimNames`).

Regarding the JWT fields, remember I mentioned in the ticket description that the spec itself points to the JWT specs when mentioning these Token Introspection response fields.